components/tabs/src/sync/store.rs

```diff
@@ -128,7 +128,15 @@ impl<'a> Store for TabsStore<'a> {
             };
             let id = record.id.clone();
             let tab = if let Some(remote_client) = self.remote_clients.borrow().get(&id) {
-                ClientRemoteTabs::from_record_with_remote_client(id, remote_client, record)
+                ClientRemoteTabs::from_record_with_remote_client(
```

Comment:
Hmm, so if we don't have an FxA device ID, it'll be the Sync clients collection ID, and a-c can't expect to find it in the FxA device list. On the other hand, `id` is already overloaded to fall back to the `tabs` record ID if we can't find that ID in `remote_clients` (because we didn't sync the clients engine, or the clients engine is out of sync with the tabs collection), and we want to nix the whole clients mechanism anyway...so it doesn't seem like it hurts anything.

Mind leaving a comment, in the Kotlin bindings, Rust, or both, about what the ID can mean, though?